pkg/operator/ceph/file/spec.go

```diff
@@ -77,54 +75,25 @@ func (c *cluster) makeDeployment(mdsConfig *mdsConfig) *apps.Deployment {
 	return d
 }
 
-func (c *cluster) makeConfigInitContainer(mdsConfig *mdsConfig) v1.Container {
-	return v1.Container{
-		Name: opspec.ConfigInitContainerName,
-		Args: []string{
-			"ceph",
-			mdsdaemon.InitCommand,
-			"--config-dir", k8sutil.DataDir,
-			"--mds-name", mdsConfig.DaemonName,
-			"--filesystem-id", c.fsID,
-			"--active-standby", strconv.FormatBool(c.fs.Spec.MetadataServer.ActiveStandby),
-		},
-		Image: k8sutil.MakeRookImage(c.rookVersion),
-		Env: []v1.EnvVar{
-			// Set '--mds-keyring' flag with an env var sourced from the secret
-			{Name: "ROOK_MDS_KEYRING",
-				ValueFrom: &v1.EnvVarSource{
-					SecretKeyRef: &v1.SecretKeySelector{
-						LocalObjectReference: v1.LocalObjectReference{Name: mdsConfig.ResourceName},
-						Key:                  keyringSecretKeyName,
-					}}},
-			k8sutil.PodIPEnvVar(k8sutil.PrivateIPEnvVar),
-			k8sutil.PodIPEnvVar(k8sutil.PublicIPEnvVar),
-			opmon.ClusterNameEnvVar(c.fs.Namespace),
-			opmon.EndpointEnvVar(),
-			opmon.SecretEnvVar(),
-			opmon.AdminSecretEnvVar(),
-			k8sutil.ConfigOverrideEnvVar(),
-		},
-		VolumeMounts: opspec.RookVolumeMounts(),
-		Resources:    c.fs.Spec.MetadataServer.Resources,
-	}
-}
-
 func (c *cluster) makeMdsDaemonContainer(mdsConfig *mdsConfig) v1.Container {
 	return v1.Container{
-		Name: "mgr",
+		Name: "mds",
 		Command: []string{
-			mdsDaemonCommand,
+			"ceph-mds",
 		},
-		Args: []string{
+		Args: append(
+			opspec.DaemonFlags(c.cephIdentity, config.MdsType, mdsConfig.DaemonName),
+			"--fsid="+c.cephIdentity.FSID(),
+			"--name=mds."+mdsConfig.DaemonName,
 			"--foreground",
-			"--id", mdsConfig.DaemonName,
-			// do not add the '--cluster/--conf/--keyring' flags; rook wants their default values
-		},
+			config.NewFlag("mds-standby-for-fscid", c.fsID),
```

Comment:
this flag is `fscid` instead of `fsid`?

Reply:
Yeah. I'm not sure what the C stands for there, but it is correct. From ceph's option code (C++):
```cpp
    Option("mds_standby_for_fscid", Option::TYPE_INT, Option::LEVEL_ADVANCED)
    .set_default(-1)
    .set_description("standby only for the file system with the given fscid"),
```

Reply:
I believe that's the RADOS context ID of the fs but I might be wrong.